modules/ec2_worker/main.tf

```diff
@@ -2,7 +2,7 @@ resource "aws_instance" "ec2_worker" {
   count = var.instance_quantity
 
   instance_type                        = var.instance_type
-  ami                                  = data.aws_ami.ec2_worker.id
+  ami                                  = var.ami_id == "" ? data.aws_ami.ec2_worker.id : var.ami_id
```

Comment:
`data.aws_ami.ec2_worker` should probably be renamed to `data.aws_ami.ec2_worker_default`, since it's not always used. Maybe it should be wrapped with a conditional `count` too, to be more efficient.